api/src/main/java/org/eclipse/microprofile/faulttolerance/Asynchronous.java

```diff
@@ -47,4 +97,15 @@
 @InterceptorBinding
 @Inherited
 public @interface Asynchronous {
+
+    public static final class CompletedFuture {
```

Comment:
What is this for? A protected method would usually return completed Future/CompletionStage, i.e. `CompletableFuture.completedFuture("hello")` from your example and throw an exception if something's wrong.

Reply:
There's no way to create a CF completed exceptionally in one line of code, as in Java 8 there's nothing like `CompletableFuture.completedFutureExceptionally(new Exception())`. Although it should be possible to just throw an exception, I wanted to add this convenience method to allow returning a CF completed exceptionally as an alternative.

Reply:
Yes, I know. But I'm not sure it's worth the effort (I think it would rarely be used) and also whether `Asynchronous` is the best place to declare such a convenient library class/method.

Reply:
OK, you're probably right that it won't be used very often. And Java 9 provides the functionality with `CF.failedFuture` so I'll remove it.